addrs/provider.go

```diff
@@ -109,6 +109,23 @@ func (pt Provider) IsZero() bool {
 	return pt == Provider{}
 }
 
+// LessThan returns true if the receiver should sort before the other given
+// address in an ordered list of provider addresses.
+//
+// This ordering is an arbitrary one just to allow deterministic results from
+// functions that would otherwise have no natural ordering. It's subject
+// to change in future.
+func (pt Provider) LessThan(other Provider) bool {
+	switch {
+	case pt.Hostname != other.Hostname:
+		return pt.Hostname < other.Hostname
+	case pt.Namespace != other.Namespace:
+		return pt.Namespace < other.Namespace
+	default:
+		return pt.Type < other.Type
+	}
+}
+
 // ParseProviderSourceString parses the source attribute and returns a provider.
```

Comment:
absolutely unrelated to this PR, but there's a similar function available in `addrs` right now, so we can switch to that one if you think it's appropriate (if so I will compare the implementations before removing this one).

Reply:
Hmm yeah... I guess this ended up here from an earlier PR before the stuff in `addrs` was ready. I'll fix this up in a subsequent PR.